File: erpnext_training_app/custom_folder/purchase_order/purchase_order.js




















































// frappe.ui.form.on("Purchase Order", {
// 	refresh(frm) {
// 		frm.trigger("prepare_custom_button");
// 	},

// 	prepare_custom_button(frm) {
// 		frm.add_custom_button(__("Make Advance Shipment Note"), () => {
// 			frappe.model.open_mapped_doc({
// 				method: "erpnext_training_app.custom_folder.purchase_order.purchase_order.make_advance_shipment_note",
// 				frm: cur_frm,
// 				freeze_message: __("Creating Advance Shipment Note ...")
// 			})
// 		});
// 	}
// });

File: erpnext_training_app/custom_folder/purchase_order/purchase_order.py
import frappe
from frappe.model.mapper import get_mapped_doc


































# @frappe.whitelist()
# def make_advance_shipment_note(source_name, target_doc=None):
# 	def update_item(source, taget, source_parent):
# 		pass

# 	doclist = get_mapped_doc(
# 		"Purchase Order",
# 		source_name,
# 		{
# 			"Purchase Order": {
# 				"doctype": "Advance Shipment Note",
# 				"validation": {"docstatus": ["=", 1]},
# 			},
# 			"Purchase Order Item": {
# 				"doctype": "Advance Shipment Note Item",
# 				"field_map": [
# 					["name", "purchase_order_item"],
# 					["parent", "purchase_order"],
# 				],
# 				"postprocess": update_item,
# 			},
# 		},
# 		target_doc,
# 	)

# 	return doclist